src/index.js

```diff
@@ -119,176 +118,115 @@ class PeerId {
   /*
    * Check if this PeerId instance is valid (privKey -> pubKey -> Id)
    */
-  isValid (callback) {
+  isValid () {
     // TODO Needs better checking
     if (this.privKey &&
       this.privKey.public &&
       this.privKey.public.bytes &&
       Buffer.isBuffer(this.pubKey.bytes) &&
       this.privKey.public.bytes.equals(this.pubKey.bytes)) {
-      callback()
+      return true
     } else {
-      callback(new Error('Keys not match'))
+      throw new Error('Keys not match')
```

Comment:
Perhaps we should take this opportunity to change this function to simply return true/false?

Reply:
Done!